src/axom/lumberjack/MPIUtility.hpp

```diff
@@ -30,6 +30,16 @@ namespace lumberjack
  */
 const char* mpiBlockingReceiveMessages(MPI_Comm comm);
 
+/*!
+ *****************************************************************************
+ * \brief Receives any Message sent to this rank, if there are any messages 
+ *  that are sent. Returns null if no messages are sent.
+ *
+ * \param [in] comm The MPI Communicator.
+ *****************************************************************************
+ */
+const char* mpiNonBlockingReceiveMessages(MPI_Comm comm, const int tag = 0);
```

Comment:
Why is `tag` argument here?

Reply:
The MPI communication calls currently use the value associated with LJ_TAG by default (defined in MPIUtility.cpp).  The non-blocking receives used by the new communicator in this PR work better when we use another tag in order to not conflict with other communicators.  I added logic into the MPI utility functions to check whether the tag was overridden (i.e. non-zero).  In those cases, the sends/receives will use the tag value passed in.  Otherwise, we revert to the default LJ_TAG for MPI communication.  Setting this default in the function declarations prevents us from having to change all the existing calls to these methods by other communicators.

Reply:
Ah. Got it. Thanks for the explanation.

Reply:
Doxygen documentation for `tag` param should be added for `mpiNonBlockingReceiveMessages`, as done for `mpiNonBlockingSendMessages`